### PR TITLE
Riverpod

### PR DESCRIPTION
**Pull request by k1mka**

### CODE REVIEW

File: lib/business_logic/dog_list_provider.dart [hunk 1]
```diff
@@ -3,19 +3,20 @@ import 'package:array_names/data/repositories/repository.dart';
 import 'package:array_names/service_locator.dart';
 import 'package:flutter_riverpod/flutter_riverpod.dart';
 
-final getDogListProvider = StateNotifierProvider(
+final getDogListProvider = StateNotifierProvider<DogListProvider, List<Breed>>(
```
Comment (AndreyKomarichev10) [inline lib/business_logic/dog_list_provider.dart:6]: JFYI, could be refactored as
`(ref) => DogListProvider( ref.read(repositoryProvider))
`  
Basically the same code
Looks better, but more complicated to understand

File: lib/business_logic/dog_list_provider.dart [hunk 2]
```diff
@@ -3,19 +3,20 @@ import 'package:array_names/data/repositories/repository.dart';
...
+    return provider;
   },
 );
 
 class DogListProvider extends StateNotifier<List<Breed>> {
   final Repository repo;
 
   DogListProvider(this.repo) : super([]);
-  getDogs() async {
+  void getDogs() async {
```
Comment (AndreyKomarichev10) [inline lib/business_logic/dog_list_provider.dart]: JFYI, could be refactored as
`
void getDogs() async => state = await repo.fetchDogs()
`

File: lib/business_logic/images_provider.dart [hunk 3]
```diff
@@ -3,7 +3,7 @@ import 'package:array_names/data/repositories/repository.dart';
 import 'package:array_names/service_locator.dart';
 import 'package:flutter_riverpod/flutter_riverpod.dart';
 
-final getImagesProvider = StateNotifierProvider(
+final getImagesProvider = StateNotifierProvider<ImagesProvider, List<String>>(
```
Comment (AndreyKomarichev10) [inline lib/business_logic/images_provider.dart:6]: 👍

File: lib/presentation/screens/images_screen/images_screen_layout.dart [hunk 4]
```diff
@@ -1,38 +1,41 @@
...
 
+  @override
+  ConsumerState<ConsumerStatefulWidget> createState() => ImagesLayoutState();
+}
+
+class ImagesLayoutState extends ConsumerState<ConsumerStatefulWidget> {
+  @override
+  void initState() {
+    ref.read(getImagesProvider.notifier).getImages(
+          (widget as ImagesLayout).breed,
```
Comment (AndreyKomarichev10) [inline lib/presentation/screens/images_screen/images_screen_layout.dart]: just a suggestion, but you could make field called parentWidget or something, assign here `widget as ImagesLayout` and there will be no need in cast every time

**Review by AndreyKomarichev10 — APPROVED** [review]

**Review by zddarova — APPROVED** [review]

**Pull request merged**